cmd/monaco/deploy/command.go

```diff
@@ -56,11 +58,26 @@ func GetDeployCommand(fs afero.Fs) (deployCmd *cobra.Command) {
 				return err
 			}
 
-			return deployConfigs(ctx, fs, manifestName, groups, environment, project, continueOnError, dryRun)
+			var limiter *rest.ConcurrentRequestLimiter
+			maxConcurrentDeployments := environment.GetEnvValueIntLog(environment.ConcurrentDeploymentsEnvKey)
+
+			if maxConcurrentDeployments > 0 {
+				log.Info("%s set, limiting concurrent deployments to %d", environment.ConcurrentDeploymentsEnvKey, maxConcurrentDeployments)
+				limiter = rest.NewConcurrentRequestLimiter(maxConcurrentDeployments)
+			}
+
+			deployOptions := deployoptions.DeployConfigsOptions{
```

Comment:
The `context` values should not be used for business logic. 
It makes sense to keep values in for logging, tracing, concurrency, etc.
Values like dry-run or continueOnErr fundamentally change how Monaco behaves. Having them in the context makes it really hard to see how code behaves.

Reply:
Since this change is quite a lot of the PR, I will have a look again after this has been changed.

Reply:
Changed 👍